### PR TITLE
New Chem: Synthetic Cocaine

### PR DESCRIPTION
**Pull request by N9neball**
## About The Pull Request
Adds a new narcotic called Synthetic Cocaine, which provides speed and stun buffs to the user as well as  
damage slowdown immunity. When overdosing, the user gets seizures as well as brain damage. When addicted, the user becomes jittery and confused, while sustaining brain loss in the later stages. The cocaine can also be placed on a tile and snorted. 
cocainum

## Why It's Good For The Game
We have meth, morphine, and bath salts, why not cocaine as well?
## Changelog
:cl:
add: New chem: Synthetic Cocaine
add: New sprite: Coke lines
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
I'm sick of the anti stun speed chems.

They're literally all just reskins of each other.

Dont' see any reason to actually add this.

**Comment by necromanceranne** [comment]
[Obligatory music for this PR](https://www.youtube.com/watch?v=oSPT27XyY1U)

Anyway, an alternative effect of the drug probably shouldn't be antistun and speedup but stamina regeneration and slowdown (one of which the drug already does). None of the current drugs deal with stamina in any way, and coke seems like a decent option for that niche.

Also slap negative moodies onto that addiction. That shit needs to make you go insane from withdrawal.

**Comment by wesoda25** [comment]
ok this is fucking cool but I hate stimulant chems they're so bad REEEEEEE

but the snorting think is soooo coool just have it make you high or something, plus damage slowdown, but have a chance to give you a heart attack every tick

**Comment by dr-bee** [comment]
We have alot of anti-stun chems already but we dont have a chem that lets you do things like build walls or craft faster. It would be a good niche for the chem, plus it would be hilarious to see assistants running around on coke speed stripping other people.

**Comment by OverlyGeneric** [comment]
just implement coke lines. another stun chem won't get in.

**Comment by Tlaltecuhtli** [comment]
give it irl effects

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -445,3 +445,69 @@
...
+	color = "#E6FFFF"
+	addiction_threshold = 10
+	overdose_threshold = 20
+
+/datum/reagent/drug/synthetic_cocaine/on_mob_life(mob/living/M)
+	var/high_message = pick("You feel like you can take on the world.", "You feel like you can walk along the razor's edge.", "You feel like you can push it to the limit.")
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	M.AdjustStun(-20, FALSE)
+	M.AdjustKnockdown(-20, FALSE)
```
Comment (Angustmeta) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: This is inconsisent with all other stun reduction chems, where all 5 reduction types have the same value.

File: code/modules/reagents/chemistry/recipes/drugs.dm [hunk 2]
```diff
@@ -47,3 +47,10 @@
 	results = list("happiness" = 4)
 	required_reagents = list("nitrous_oxide" = 2, "epinephrine" = 1, "ethanol" = 1)
 	required_catalysts = list("plasma" = 5)
+	
+/datum/chemical_reaction/synthetic_cocaine
+	name = "Synthetic Cocaine"
+	id = "synthetic_cocaine"
+	results = list("synthetic_cocaine" = 4)
+	required_reagents = list("ephedrine" = 1, "ammonia" = 1, "welding_fuel" = 1, "nitrogen" = 1)
```
Comment (Angustmeta) [inline code/modules/reagents/chemistry/recipes/drugs.dm]: This is easily made with a single macro+heating. Ignoredamageslowdown is a powerful trait. Would be more interesting if it used one of the lavaland mushroom chems IMO.

**Comment by Angustmeta** [comment]
Take cocaine. All the time. 

Jokes aside, I think this is too OP in its current form. 

I don't see any speed up anywhere and I don't think you should add it. We have enough stimulants. This might be a fun combat/grief tool though by creating decals that insta-OD people people who accidentally click on them with 20u cocaine, if that's an intended use. 

**Suggestions:**

- Don't add the speed boost, but keep ignoredamageslowdown since that feels thematic with the overperforming workers on cocaine meme. 
- Add a new downside: Not being able to tell how hurt you are (screwyhud from miner's salve? Or is there a trait now?). Same theme. 
- Make snorting cocaine decals take a second. Give them like 15u instead of 20. 
- Change the stun reductions to like 20 instead for consistency. 
- Change the recipe from using ephedrine to using **[Entropic Polypnium](https://tgstation13.org/wiki/Guide_to_chemistry#Entropic_Polypnium)**. Because this is fun, gives a use to the lavaland tall mushroom, and would make the drug safe to add to the game without making it another macro drug to print with dispenser. It would rarely be made but I think that's fine. It would be an alternative for miners who can't find enough legion cores for megafauna, to bring up to the chemists. Might even be slightly underpowered.

**Comment by 81Denton** [comment]
The line snorting mechanic is pretty cool, but as other people pointed out we already have a bunch of reskinned antistun chems that all pretty much do the same thing.
Coke will have a better chance of getting merged if you change the effects to something unique, like the faster toolspeed that @dr-bee mentioned.

**Comment by Grazyn** [comment]
A sound effect for the snorting would be cool

**Comment by FantasticFwoosh** [comment]
So i can't just play det so i can swab my finger through 1u of it, taste and identify like a DEA agent.

_Deep in thought, the detective dials in the comm channel - "Sir Its wheat, not cocaine"_

My immersion is broken so hard right now

**Comment by ATH1909** [comment]
So you can dupe synthetic cocaine by throwing 15u of it on the ground, then collecting 20u of it from that pile?

Why not make the amount of cocaine in the pile based on the amount of cocaine that you use to make the pile? Change the snorting interaction with the pile to just transfer chems from the pile into you (maybe as a vapor reaction?) so you can poison/mix other drugs into cocaine piles or something (and also snort "safe" amounts of cocaine by putting less than 20u in a pile).

**Comment by Wilchenx** [comment]
>adding main negatives to being addicted

Bad mechanic that we should stop having

make withdrawal be twice as painful as being addicted now, and reduce effects of being addicted

**Comment by ATH1909** [comment]
> > adding main negatives to being addicted
> 
> Bad mechanic that we should stop having
> 
> make withdrawal be twice as painful as being addicted now, and reduce effects of being addicted

So we should move the major negatives of being addicted to drugs to the later stages of those addictions? Sounds like a good idea, and I think we (and by "we", I mean "someone other than me, because I'm lazy and haven't bothered to figure out how to make PRs") should definitely make a PR for that.

**Comment by Sisyphusrollson** [comment]
I appreciate the feedback. I will be making some changes to the chem later.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 3]
```diff
@@ -200,3 +200,21 @@
...
+	. = ..()
+	if(.)
+		return
+	if(ishuman(M))
+		M.reagents.add_reagent("synthetic_cocaine", 20)
+		to_chat(M, "<span class='notice'>You snort the cocaine.</span>")
+		Destroy()
+/obj/effect/decal/cleanable/coke/Initialize()
+	. = ..()
+	reagents.add_reagent("synthetic_cocaine", 20)
```
Comment (Shardj) [inline code/game/objects/effects/decals/cleanable/misc.dm]: You're missing an end of file newline

Comment (optimumtact) [reply code/game/objects/effects/decals/cleanable/misc.dm]: doesn't matter

**Comment by optimumtact** [comment]
This is straight up bloat

**Pull request closed without merging**